modules/core/perf/opencl/perf_arithm.cpp

```diff
@@ -678,7 +678,12 @@ OCL_PERF_TEST_P(SqrtFixture, Sqrt, ::testing::Combine(
 
     OCL_TEST_CYCLE() cv::sqrt(src, dst);
 
-    if (CV_MAT_DEPTH(type) >= CV_32F)
+    // To square root 32 bit floats we use native_sqrt, which has implementation
+    // defined accuracy. We know intel devices have accurate native_sqrt, but
+    // otherwise stick to a relaxed sanity check. For types larger than 32 bits
+    // we can do the accuracy check for all devices as normal.
+    if ((CV_MAT_DEPTH(type) == CV_32F && ocl::Device::getDefault().isIntel()) ||
```

Comment:
Need to properly handle case with disabled OpenCL (use strict variant of check tolerance):
```
#ifdef HAVE_OPENCL
bool strictCheck = !ocl::useOpenCL() || ocl::Device::getDefault().isIntel();
#else
bool strictCheck = true;
#endif
if (... strictCheck ...)
    strict;
else
    relax;
```

Reply:
looks like this whole file is wrapped in `#ifdef HAVE_OPENCL` https://github.com/opencv/opencv/blob/4d2e035902e0b6ac1695ea4d1571d294dfd74db2/modules/core/perf/opencl/perf_arithm.cpp#L45

Reply:
`ocl::useOpenCL()` check is still necessary. `#ifdef HAVE_INTEL` doesn't help in all cases (compile with OpenCL, but run with `OPENCV_OPENCL_RUNTIME=disabled`)

Reply:
Aahh makes sense, sorry about that. Should be fixed now